### PR TITLE
[CHN336]-add textarea support for km-input-lstring

### PR DESCRIPTION
**Pull request by liyuanmontreal**
add one props rows for distinguish  show input or textarea
    rows=1, show input
    rows>1, show textarea
change examples in preview page,
   show modelValue below the example
   add one example for textarea
image

### CODE REVIEW

File: src/components/controls/km-input-lstring.vue [hunk 1]
```diff
@@ -1,30 +1,14 @@
...
-        class="form-control"
-        aria-describedby="basic-addon2"
-        v-model="model[locale]"
-        :dir="locale == 'ar' ? 'rtl' : 'ltr'"
-        @input="$emit('update:modelValue', model)"
-      />
+    <div class="input-group mb-1" v-for="locale in locales" :key="locale" :class="`km-input-lstring-${locale}`">
+      <input v-if="rows==1" type="text" class="form-control" aria-describedby="basic-addon2" v-model="model[locale]" :dir="locale == 'ar' ? 'rtl' : 'ltr'" @input="$emit('update:modelValue', model)">
+      </input>
+      <textarea  v-if="rows>1"  class="form-control" placeholder=""  v-model="model[locale]" :dir="locale == 'ar' ? 'rtl' : 'ltr'" @input="$emit('update:modelValue', model)">
```
Comment (blaisefonseca) [inline src/components/controls/km-input-lstring.vue]:
```suggestion
      <textarea  v-if="rows>1"  class="form-control" placeholder=""  v-model="model[locale]" :dir="locale == 'ar' ? 'rtl' : 'ltr'" @input="$emit('update:modelValue', model)" :rows="rows>        
```

File: src/components/controls/km-input-lstring.vue [hunk 2]
```diff
@@ -38,8 +22,9 @@ import { without } from "lodash";
 
 const model = defineModel({type:Object, required: true, default:{}})
 const props = defineProps({
-  locales: { type: Array, required: true },
+  locales:  { type: Array, required: true },
   disabled: { type: Boolean, default: false },
+  rows:     { type: Number, required: false, default: 1 }
```
Comment (blaisefonseca) [inline src/components/controls/km-input-lstring.vue]: can you add placeholder property as well

File: src/components/controls/km-input-lstring.vue [hunk 3]
```diff
@@ -1,30 +1,14 @@
...
-        type="text"
-        class="form-control"
-        aria-describedby="basic-addon2"
-        v-model="model[locale]"
-        :dir="locale == 'ar' ? 'rtl' : 'ltr'"
-        @input="$emit('update:modelValue', model)"
-      />
+    <div class="input-group mb-1" v-for="locale in locales" :key="locale" :class="`km-input-lstring-${locale}`">
+      <input v-if="rows==1" type="text" class="form-control" aria-describedby="basic-addon2" v-model="model[locale]" :dir="locale == 'ar' ? 'rtl' : 'ltr'" @input="$emit('update:modelValue', model)">
+      </input>
```
Comment (blaisefonseca) [inline src/components/controls/km-input-lstring.vue]:
```suggestion
```
```suggestion
      <input v-if="rows==1" type="text" class="form-control" aria-describedby="basic-addon2" v-model="model[locale]" :dir="locale == 'ar' ? 'rtl' : 'ltr'" @input="$emit('update:modelValue', model)" />
```

File: src/components/controls/km-input-lstring.vue [hunk 4]
```diff
@@ -1,30 +1,14 @@
 <template>
   <div :id="$attrs.id" class="scbd-controls km-input-lstring" :class="$attrs.class">
```
Comment (blaisefonseca) [inline src/components/controls/km-input-lstring.vue]:
```suggestion
  <div  class="scbd-controls km-input-lstring" >
```

Comment (blaisefonseca) [reply src/components/controls/km-input-lstring.vue]: https://vuejs.org/guide/components/attrs

**Comment by liyuanmontreal** [comment]
fix above code review issues.
image

File: src/components/controls/km-input-lstring.vue (hunk 4 above)
Comment (liyuanmontreal) [reply src/components/controls/km-input-lstring.vue]: done

File: src/components/controls/km-input-lstring.vue (hunk 3 above)
Comment (liyuanmontreal) [reply src/components/controls/km-input-lstring.vue]: done

File: src/components/controls/km-input-lstring.vue (hunk 2 above)
Comment (liyuanmontreal) [reply src/components/controls/km-input-lstring.vue]: done

File: src/components/controls/km-input-lstring.vue (hunk 1 above)
Comment (liyuanmontreal) [reply src/components/controls/km-input-lstring.vue]: done

**Review by blaisefonseca — APPROVED** [review]

**Pull request merged**